packages/wallet/src/cip30.ts

```diff
@@ -154,14 +197,20 @@ export const createWalletApi = (
   },
   submitTx: async (tx: Cbor): Promise<string> => {
     logger.debug('submitting tx');
-    try {
-      const txDecoded = CSL.Transaction.from_bytes(Buffer.from(tx, 'hex'));
-      const txData: Cardano.NewTxAlonzo = cslToCore.newTx(txDecoded);
-      await wallet.submitTx(txData);
-      return Promise.resolve(txData.id.toString());
-    } catch (error) {
-      logger.error(error);
-      throw error;
+    const txDecoded = CSL.Transaction.from_bytes(Buffer.from(tx, 'hex'));
+    const txData: Cardano.NewTxAlonzo = cslToCore.newTx(txDecoded);
+    const shouldProceed = await confirmationCallback({ data: txData, type: Cip30ConfirmationCallbackType.SubmitTx });
```

Comment:
Actually this callback should be called in signTx. Don't think we need to guard against submitting. This will also change the data type from NewTxAlonzo to either NewTxBodyAlonzo or TxInternals

Reply:
Don't we need both?
**SubmitTx**
> If the wallet accepts the transaction and tries to send it,
**SignTx**
> The wallet should ask the user for permission,

Sounds like `signData`, `signTx`, and `submitTx` should all have confirmation callbacks for them as they are user-led actions, what do you think to me adding another instead of simply switching?

Reply:
Yeah adding a another sounds right